curriculum/challenges/english/01-responsive-web-design/learn-css-colors-by-building-a-color-markers-set/step-050.md

```diff
@@ -0,0 +1,89 @@
+---
+id: 61a493ead935c148d2b76312
+title: Step 50
+challengeType: 0
+dashedName: step-50
+---
+
+# --description--
+
+The `linear-gradient` function is very flexible -- here is the basic syntax you'll use in this tutorial:
+
+```css
+linear-gradient(gradientDirection, color1, color2, ...);
+```
+
+`gradientDirection` is the direction of the line used for the transition. `color1` and `color2` are color arguments, which are the colors that will be used in the transition itself. These can be any type of color, including color keywords, hex, `rgb`, or `hsl`.
+
+Now you'll apply a red-to-green gradient along a 90 degree line to the first marker.
+
+First, in the `.red` CSS rule, set the `background` property to `linear-gradient()`, and pass it the value `90deg` as the `gradientDirection`.
+
+# --hints--
+
+Your `.red` CSS rule should have a `background` property with the value `linear-gradient(90deg)`.
+
+```js
+assert(
+  __helpers.removeWhiteSpace($('style').text()).match(/\.red{.*background:linear-gradient\(90deg\).*}/gi)
+);
```

Comment:
I did not previously see the jQuery. As another PS:
The `getPropVal` method works almost identically to `getPropertyValue`, but we modified it to take a boolean as a second argument indicating whether to remove whitespace or not.

This is done for two reasons:
1) As I am sure you discovered, with CSS values using CSS functions, whitespace is included, for some unknown reason. So, you can remove it.
2) Do not use jQuery, because, technically, this code will pass the tests:
```
.red {
  back grou nd: lin ear- gradient (9 0 de g)
}
```

The reason the CSSHelp function does not allow that to pass is because it works with _computed_ styles. So, if V8 cannot compute the value put in, then it will not create a style for it.

_Extreme. But you get the point._

```suggestion
assert.equal(__helpers.CSSHelp(document).getStyle('.red').getPropVal('background', true), 'linear-gradient(90deg)')
```